pennylane/operation.py

```diff
@@ -962,15 +962,6 @@ def kraus_matrices(self):
         return self._kraus_matrices(*self.parameters)
 
     def __init__(self, *params, wires=None, do_queue=True):
-
-        # check parameters are valid
-        if self.par_domain == "R" and any(not 0 <= np.real(p) <= 1 for p in params):
-            raise ValueError("Channel probability parameters should be numbers between 0 and 1.")
```

Comment:
Out of curiosity, how come this was removed?

Reply:
Nevermind, I re-read your PR comment and now see why.

Two questions:

* What currently happens if `p` outside this range is given? Is there an obscure error? Does it work, but provide results that are unphysical (like when you have too low a cutoff in the Fock backends?)

* I support we could add a `qml.math.clip` function, and then we could clip values between 0-1. This would allow optimization to work and remain physical.

Reply:
We get an error anyway from constructing the Kraus matrices:
```
RuntimeWarning: invalid value encountered in sqrt
  K0 = np.sqrt(1 - p) * np.eye(2)
```

I could add a check again, I'm just not sure how it works when the parameters are variables. I think it's fine not to include the check

Reply:
Maybe a case like `if isinstance(p, qml.variable.Variable)` and then checking `var.val` could help (should the check still be required).

Reply:
From a user standpoint I think having the original error message is more clear because it specifies what is wrong with the input at a conceptual level, rather than an invalid value just getting caught in the sqrt.